R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,23 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  # do not download if it is run in the sparkR shell
+  if (!grepl(".*shell\\.R$", Sys.getenv("R_PROFILE_USER"), perl = TRUE)) {
+    if (!nzchar(master) || is_master_local(master)) {
```

Comment:
Ah, since the install function is restricted to local for now, perhaps we only have to do the check for local master? If so, then I think it would be better to flip the order of `if` statements...